mnefun/_ssp.py

```diff
@@ -285,7 +285,7 @@ def do_preprocessing_combined(p, subjects, run_indices):
                   % op.sep.join(bad_file.split(op.sep)[-3:]))
             bad_file = None
 
-        ecg_t_lims = p.ecg_t_lims
+        ecg_t_lims = _handle_dict(p.ecg_t_lims, subj)
```

Comment:
These limits are supposed to be floats so `mne` doesn't balk. Are we going to convert everything to `dict` to get this `yaml` business to work? I've had to flip flop 10-20 parameters from `yaml` to explicit `mnefun.params` expressions just to get through one subject through `SSP` and now I have a 100 lines long YAML file and a 100 lines long MNEFUN script. How is this better, again? @NeuroLaunch

Reply:
If you look at what `_handle_dict` [does](https://github.com/LABSN/mnefun/blob/master/mnefun/_utils.py#L74-L81):
```
def _handle_dict(entry, subj):
    out = entry
    if isinstance(entry, dict):
        try:
            out = entry[subj]
        except KeyError:
            pass
    return out
```
Hopefully you would agree that it will not break things if `p.ecg_tlims` is a float. Basically it allows dict-ness but does not force it.

> I've had to flip flop 10-20 parameters from yaml to explicit mnefun.params expressions just to get one subject through SSP

This should not be the case, something is either wrong with the `mnefun` code or how you are translating your parameters. If you can fix the `badbaby` code to use the single, non-looped path (I would consider this a prerequisite for converting to YAML) and then open a WIP PR to convert to YAML then I can comment on where the issues are. If this is a pain in terms of GitHub gymnastics, I'm happy to "put my changes back" in one PR that makes a single flat `py` file with no loops, then do a draft to convert it to YAML conversion in another.

This latter part is basically what this PR was, the translation wasn't too bad, and the `analysis_mnefun.py` script ends up being very short:

https://github.com/LABSN/mnefun/pull/270/files#diff-7aaebcfef6318ee3bec53bec525f8b3b39ef1e2fedce4fbbc8daef3db3aed939